dex/src/main/java/io/crate/data/InMemoryBatchIterator.java

```diff
@@ -51,6 +51,8 @@
     }
 
     /**
+     * @param items An iterable over the items. Depending on {@code RowConsumer#requiresRepeat()},
```

Comment:
I don't think this comment makes sense.
`repeatable` in the context of an `Iterable` is a bit strange and looking at this purely from a `BatchIterator` and `Iterable` perspective the contract is that the `BatchIterator` will only produce the same results after `moveToStart` if each `iterator` call on the `Iterable` produces the same results. This has no direct relationship with `RowConsumer#requiresRepeat`. It's just one case where the consumer of the BatchIterator happens to use `moveToStart`

Reply:
Thanks for clarifying. I understand now how the components work together.